packages/react-hooks/src/useBlockInterval.ts

```diff
@@ -40,10 +42,16 @@ function calcInterval (api: ApiPromise): BN {
 function useBlockIntervalImpl (apiOverride?: ApiPromise | null): BN {
   const { api } = useApi();
 
+  const currApi = apiOverride || api;
+  const blockTimeAura = useCall(currApi.call.auraApi?.slotDuration && currApi.call.auraApi.slotDuration, []);
+  const blockTimeBabe = useCall(currApi.call.babeApi?.configuration && currApi.call.babeApi.configuration, [], {
+    transform: (data: BabeGenesisConfiguration | undefined) => data?.slotDuration
+  });
+
   return useMemo(
-    () => calcInterval(apiOverride || api),
-    [api, apiOverride]
-  );
+    () => (blockTimeAura || blockTimeBabe) ?? calcInterval(currApi),
+    [blockTimeAura, blockTimeBabe, currApi]
+  ) as BN;
```

Comment:
Then with the above you can remove the `as BN`.

Thanks so much for the patience. I just realized we can remove the typecasting here by using the Generic input above.

Reply:
Thanks to you for the feedback!

Addressed in [0e593af](https://github.com/polkadot-js/apps/pull/10457/commits/0e593afd5d5c90d6b1d2cd36eee770dc647a9138).